PurchasesCoreSwift/Public/EntitlementInfo.swift

```diff
@@ -0,0 +1,346 @@
+           let unwrapped = maybeSandbox as? NSNumber {
+            isSandbox = unwrapped.boolValue
+        } else if maybeSandbox?.responds(to: #selector(getter: NSString.boolValue)) ?? false,
+                  let unwrapped = maybeSandbox as? NSString {
+            isSandbox = unwrapped.boolValue
+        } else {
+            isSandbox = false
+        }
+
+        self.isSandbox = isSandbox
+        self.isActive = Self.isDateActive(expirationDate: entitlementExpiresDate, forRequestDate: requestDate)
+        self.periodType = Self.parsePeriodType(periodType: periodTypeString)
+        self.latestPurchaseDate = Self.parseDate(dateString: entitlementPurchaseDateString,
+                                                 withDateFormatter: dateFormatter)
+        self.originalPurchaseDate = Self.parseDate(dateString: originalPurchaseDateString,
+                                                   withDateFormatter: dateFormatter)
+        self.ownershipType = Self.parseOwnershipType(ownershipType: ownershipType)
+        self.willRenew = Self.willRenewWithExpirationDate(expirationDate: expirationDate,
+                                                          store: store,
+                                                          unsubscribeDetectedAt: unsubscribeDetectedAt,
+                                                          billingIssueDetectedAt: billingIssueDetectedAt)
+    }
+
+    class func parseDate(dateString: String?, withDateFormatter dateFormatter: DateFormatter) -> Date? {
+        guard let dateString = dateString else {
+            return nil
+        }
+
+        return dateFormatter.date(from: dateString)
+    }
+
+    class func isDateActive(expirationDate: Date?, forRequestDate requestDate: Date?) -> Bool {
+        guard let expirationDate = expirationDate else {
+            return true
+        }
+
+        let referenceDate: Date = requestDate ?? Date.init()
+        return expirationDate.timeIntervalSince(referenceDate) > 0
+    }
+
+    class func parseOwnershipType(ownershipType: String?) -> PurchaseOwnershipType {
+        switch ownershipType {
+        case nil:
+            return .purchased
+        case "PURCHASED":
+            return .purchased
+        case "FAMILY_SHARED":
+            return .familyShared
+        default:
+            // TODO(post-migration check): Logging?
+            return .unknown
+        }
+    }
+
+    class func parsePeriodType(periodType: String?) -> PeriodType {
+        switch periodType {
+        case "normal":
+            return .normal
+        case "intro":
+            return .intro
+        case "trial":
+            return .trial
+        default:
+            // TODO(post-migration check): Also handles nil.
+            return .normal
+        }
+    }
+
+    class func parseStore(store: String?) -> Store {
+        switch store {
+        case "app_store":
+            return .appStore
+        case "mac_app_store":
+        return .macAppStore
```

Comment:
we're missing indentation here. 
Should the linter have caught this?

Reply:
This is actually default Swift indentation. I hate it. But it is `correct`. Sigh.

Reply:
Wait, I see what you mean. I'm wrong.